src/find/mod.rs

```diff
@@ -854,4 +854,112 @@ mod tests {
+                let arg = &format!("-newer{}{}", x, y).to_string();
+                let deps = FakeDependencies::new();
+                let rc = find_main(
+                    &[
+                        "find",
+                        "./test_data/simple/subdir",
+                        arg,
+                        "./test_data/simple/subdir/ABBBC",
+                    ],
+                    &deps,
+                );
+
+                assert_eq!(rc, 0);
+            });
+        });
+    }
+
+    #[test]
+    #[cfg(target_os = "linux")]
+    fn test_find_newer_xy_have_not_birthed_time_filesystem() {
+        let y_options = ["a", "c", "m"];
+        y_options.iter().for_each(|&y| {
+            let arg = &format!("-newerB{}", y).to_string();
+            let deps = FakeDependencies::new();
+            let rc = find_main(
+                &[
+                    "find",
+                    "./test_data/simple/subdir",
+                    arg,
+                    "./test_data/simple/subdir/ABBBC",
+                ],
+                &deps,
+            );
+
+            assert_eq!(rc, 1);
+        });
+    }
+
+    #[test]
+    fn test_find_newer_xy_before_created_time() {
+        // normal - before the created time
+        #[cfg(target_os = "linux")]
+        let args = ["-newerat", "-newerct", "-newermt"];
+        #[cfg(not(target_os = "linux"))]
+        let args = ["-newerat", "-newerBt", "-newerct", "-newermt"];
+        let times = ["jan 01, 2000", "jan 01, 2000 00:00:00"];
+
+        for (arg, time) in args.iter().zip(times.iter()) {
+            let deps = FakeDependencies::new();
+            let rc = find_main(&["find", "./test_data/simple/subdir", arg, time], &deps);
+
+            assert_eq!(rc, 0);
+            assert_eq!(
+                deps.get_output_as_string(),
+                fix_up_slashes("./test_data/simple/subdir\n./test_data/simple/subdir/ABBBC\n"),
+            );
+        }
+    }
+
+    #[test]
+    fn test_find_newer_xy_after_created_time() {
+        // normal - after the created time
+        #[cfg(target_os = "linux")]
+        let args = ["-newerat", "-newerct", "-newermt"];
+        #[cfg(not(target_os = "linux"))]
+        let args = ["-newerat", "-newerBt", "-newerct", "-newermt"];
+        let times = ["jan 01, 2037", "jan 01, 2037 00:00:00"];
+
+        for (arg, time) in args.iter().zip(times.iter()) {
+            let deps = FakeDependencies::new();
+            let rc = find_main(&["find", "./test_data/simple/subdir", arg, time], &deps);
+
+            assert_eq!(rc, 0);
+            assert_eq!(deps.get_output_as_string(), "");
+        }
+    }
+
+    #[test]
+    fn test_find_newer_xy_error_path() {
+        // Catch a parsing error.
+        #[cfg(target_os = "linux")]
+        let args = ["-newerat", "-newerct", "-newermt"];
+        #[cfg(not(target_os = "linux"))]
+        let args = ["-newerat", "-newerBt", "-newerct", "-newermt"];
+        let times = [""];
```

Comment:
You don't need an array for a single value. Using `let time = "";` allows you to simplify the `for` loop below.